go/tools/common/common.go

```diff
@@ -17,19 +17,38 @@ limitations under the License.
 package common
 
 import (
-	"log"
+	"fmt"
+	"path"
+	"strings"
 
 	"golang.org/x/tools/go/packages"
 )
 
-// PkgFailed returns true if any of the packages contain errors
-func PkgFailed(loaded []*packages.Package) bool {
-	failed := false
-	for _, pkg := range loaded {
-		for _, e := range pkg.Errors {
-			log.Println(e.Error())
-			failed = true
+func CheckErrors(loaded []*packages.Package, skip func(fileName string) bool) error {
+	var errors []string
+	for _, l := range loaded {
+		for _, e := range l.Errors {
+			idx := strings.Index(e.Pos, ":")
+			filePath := e.Pos[:idx]
+			_, fileName := path.Split(filePath)
+			if !skip(fileName) {
+				errors = append(errors, e.Error())
+			}
 		}
 	}
-	return failed
+	if len(errors) > 0 {
+		return fmt.Errorf("found %d error(s) when loading Go packages:\n\t%s", len(errors), strings.Join(errors, "\n\t"))
```

Comment:
Guess we're almost on Go 1.20 where we could use `errors.Join` instead.  Not as important here though as this is not something where the caller would want to dig into the actual different separate error messages, as it's all printed to the command line for the human running this afaik?